api/WebApi.ts

```diff
@@ -366,12 +366,20 @@ export class WebApi {
         if (!process.env.no_proxy) {
             return false;
         }
+
         const noProxyDomains = (process.env.no_proxy || '')
         .split(',')
         .map(v => v.toLowerCase());
         const serverUrl = url.parse(_url).host.toLowerCase();
-        // return true if the no_proxy includes the host
-        return noProxyDomains.indexOf(serverUrl) !== -1;
+
+        for (const domain of noProxyDomains) {
+            // if the domain of the server is listed, return true
+            if (serverUrl.endsWith(domain)) {
```

Comment:
I think this has a bug, we would want to match when it's `.example.com` but not when it's `example.com`. This code would match on both.

Reply:
We should add more tests for this.